src/NicBaseSelector.re

```diff
@@ -0,0 +1,134 @@
+type nic = {
+  concentration: int,
+  base: Constants.base,
+  index: int,
+};
+
+[@react.component]
+let make = (~handleSetNicWeight) => {
+  let nicOptions: list(nic) = [
+    {concentration: 100, base: Constants.PG, index: 0},
```

Comment:
`index` is a hack, i didn't find a way to get the index from `List.map`

Reply:
You can use `List.mapi`, which passes the index as the first argument to your map function: https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html#VALmapi